### ISSUE TITLE
Not calculating best plan after 8.1.0

### ISSUE THREAD

**Issue by Philhews** [springfall2008/batpred#1284]
**Describe the bug**
Tomorrows agile rates are negative in the afternoon, and the solar forecast is not enough to max out my inverter so freeze discharging before and using some negative grid to charge later is the best cost. This is the plan that is selected on 8.1.0

However when upgrading past that version (anything up to latest) the plan no longer utilises the AC negative rates and just does solar charge with a demonstrably worse 'best' cost value.

**Expected behavior**
Best plan should be selected on newer versions

**Predbat version**
8.1.0

**Environment details**

Hybrid 3.6 Gen2, 9.5kWh battery
RPI4 with HAOS and standalone predbat

**Screenshots**
Plan on 8.1.0
![image](https://github.com/springfall2008/batpred/assets/117396644/12c8aa33-05db-437d-ba43-7f93a65270af)
Cost on 8.1.0
![image](https://github.com/springfall2008/batpred/assets/117396644/30b8d07f-6d0f-4b3d-abf4-b78e28cfb24d)
Battery graph on 8.1.0
![image](https://github.com/springfall2008/batpred/assets/117396644/50bea18f-16ae-4850-8922-1d947eee4d60)

Plan on 8.1.1
![image](https://github.com/springfall2008/batpred/assets/117396644/fb4c87df-6a40-49c8-94b3-6630b124eb93)
Cost on 8.1.1
![image](https://github.com/springfall2008/batpred/assets/117396644/777112aa-a14c-4bb0-8d95-4dbe20ac6bce)
Battery graph on 8.1.1
![image](https://github.com/springfall2008/batpred/assets/117396644/0f14985b-bce8-4953-8aa9-325414db3767)

**Log file**
Update from 8.1.0 to 8.1.1 happened at 20:53 so log from before then is working as I would expect, and immediately after is the problem
[predbat.log](https://github.com/user-attachments/files/16113382/predbat.log)

**Comment by Philhews** [comment]
I'll see how that affects the current plans, obviously the rates are different now so it's not a direct comparison.

Did something change between the two versions mentioned that would make these settings more/less important in the calculation to result in the differences I was seeing?

**Comment by Philhews** [comment]
8.1.1 still tending towards less charging, requires more solar to refill the battery rather than using grid at 7 or 9p

8.1.0 plan
![Screenshot_2024-07-06-23-01-25-73_c3a231c25ed346e59462e84656a70e50](https://github.com/springfall2008/batpred/assets/117396644/087b67e0-2ab6-429a-9999-d25d87e12dbc)

8.1.1 plan
![Screenshot_2024-07-06-23-05-14-50_c3a231c25ed346e59462e84656a70e50](https://github.com/springfall2008/batpred/assets/117396644/e48a8ded-2d09-4cae-b531-e07342c820aa)

Log file, update to .1 at 23.02
[predbat.log](https://github.com/user-attachments/files/16117499/predbat.log)

**Comment by springfall2008** [comment]
> I'll see how that affects the current plans, obviously the rates are different now so it's not a direct comparison.
> 
> Did something change between the two versions mentioned that would make these settings more/less important in the calculation to result in the differences I was seeing?

Yes, there were bugs in the code which mean it wasn't always applying losses correctly.

**Comment by springfall2008** [comment]
As this ticket is old I'm going to close it, please re-open if the issue is there on latest.
